Rakefile

```diff
@@ -0,0 +1,63 @@
+require "rubygems"
+require 'rake'
+require 'yaml'
+require 'time'
+
+SOURCE = "."
+CONFIG = {
+	'version_file' => File.join(SOURCE, '_version.yml'),
+	'layouts' => File.join(SOURCE, '_layouts'),
+	'posts' => File.join(SOURCE, '_posts'),
+	'post_ext' => 'md',
+}
+
+def agree?(message)
+	valid_options = [ 'y', 'n' ]
+	answer = get_stdin("#{message} ").downcase[0]
+
+	if answer == 'y'
+		true
+	else
+		false
+	end
+end
+
+def get_stdin(message)
+  print message
+
+  STDIN.gets.chomp
+end
+
+desc "Begin a new post in #{CONFIG['posts']}"
+task :post do
+	abort("rake aborted: '#{CONFIG['posts']}' directory not found.") unless FileTest.directory?(CONFIG['posts'])
+	title = ENV["title"] || "new-post"
+	tags = ENV["tags"] || "[]"
+	slug = title.downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, '')
+
+	begin
+		date = (ENV['date'] ? Time.parse(ENV['date']) : Time.now).strftime('%Y-%m-%d')
+	rescue => e
+		puts "Error - date format must be YYYY-MM-DD, please check you typed it correctly!"
+		exit -1
+	end
+
+	filename = File.join(CONFIG['posts'], "#{date}-#{slug}.#{CONFIG['post_ext']}")
+	if File.exists?(filename)
+		abort('Rake task aborted.') if agree?("#{filename} already exists. Do you want to override it?") == false
```

Comment:
It may be worth using `unless` to tidy up the statement:

```diff
-		abort('Rake task aborted.') if agree?("#{filename} already exists. Do you want to override it?") == false 
+		abort('Rake task aborted.') unless agree?("#{filename} already exists. Do you want to override it?")

```